registrations/tests/test_send_message.py

```diff
@@ -60,77 +63,114 @@ def test_email_is_sent_to_responsible_signups(api_client, registration, user):
         responsible_for_group=True,
         email="test2@test.com",
     )
-    SignUpFactory(
-        signup_group=signup_group, registration=registration, email="test3@test.com"
-    )
+
+    # Individual
+    third_signup = SignUpFactory(registration=registration, email="test3@test.com")
+    fourth_signup = SignUpFactory(registration=registration, email="test3@test.com")
 
     api_client.force_authenticate(user)
     send_message_data = {"subject": "Message subject", "body": "Message body"}
 
     assert_send_message(
-        api_client, registration.id, send_message_data, [second_signup.email]
+        api_client,
+        registration.id,
+        send_message_data,
+        [
+            first_signup.email,
+            second_signup.email,
+            third_signup.email,
+            fourth_signup.email,
+        ],
     )
     # Default language for the email is Finnish
     assert "Tarkastele ilmoittautumistasi täällä" in str(mail.outbox[0].alternatives[0])
 
 
 @pytest.mark.django_db
-def test_email_is_sent_to_group_signups_without_responsible_signups(
-    api_client, registration, user
-):
-    signup_group = SignUpGroupFactory(registration=registration)
-    first_signup = SignUpFactory(
-        signup_group=signup_group, registration=registration, email="test@test.com"
+def test_email_is_sent_to_selected_signup_groups_only(api_client, registration, user):
+    signup_group0 = SignUpGroupFactory(registration=registration)
+    SignUpFactory(
+        signup_group=signup_group0, registration=registration, email="test@test.com"
     )
-    second_signup = SignUpFactory(
-        signup_group=signup_group, registration=registration, email="test2@test.com"
+    SignUpFactory(
+        signup_group=signup_group0,
+        registration=registration,
+        responsible_for_group=True,
+        email="test2@test.com",
     )
+
+    signup_group1 = SignUpGroupFactory(registration=registration)
     third_signup = SignUpFactory(
-        signup_group=signup_group, registration=registration, email="test3@test.com"
+        signup_group=signup_group1,
+        registration=registration,
+        responsible_for_group=True,
+        email="test3@test.com",
+    )
+    SignUpFactory(
+        signup_group=signup_group1, registration=registration, email="test4@test.com"
     )
 
     api_client.force_authenticate(user)
-    send_message_data = {"subject": "Message subject", "body": "Message body"}
+    send_message_data = {
+        "subject": "Message subject",
+        "body": "Message body",
+        "signup_groups": [signup_group1.pk],
+    }
 
     assert_send_message(
-        api_client,
-        registration.id,
-        send_message_data,
-        [first_signup.email, second_signup.email, third_signup.email],
+        api_client, registration.id, send_message_data, [third_signup.email]
```

Comment:
The name of the test is `test_email_is_sent_to_selected_signup_groups_only`; shouldn't the email be sent to `test4@test.com` as well then?

Reply:
Improved the test case name in the commit 34128889b16ed30ae20ff09e74526edd7565e798. It was otherwise correct that `test4@test.com` shouldn't get the email as it isn't a `responsible_for_group` signup (and since the group already has the third signup marked as the responsible one).